### ISSUE TITLE
Calculate loss at beginning and end of training

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1221]
I talked to some users who were a bit confused / surprised that the validation loss didn't get calculated during a run. I think that's a valid issue if you have a small dataset where your training steps are smaller than the default `eval.interval=100`.

I think the best way to address this is to also always calculate the validation loss always at the beginning and end of each training run. These would also be useful reference points to have in general. 

Should be easy for me to add.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — I think this can be closed as already shipped: both an initial-validation pass and a final-validation pass are now first-class `EvalArgs` options.

Evidence:
- The flags live in [`litgpt/args.py:87,89`](https://github.com/Lightning-AI/litgpt/blob/main/litgpt/args.py#L87-L89):
  ```python
  initial_validation: bool = False
  ...
  final_validation: bool = True
  ```
- They are wired into every training script — e.g. in [`litgpt/finetune/lora.py:310`](https://github.com/Lightning-AI/litgpt/blob/main/litgpt/finetune/lora.py#L310) (initial) and [L260](https://github.com/Lightning-AI/litgpt/blob/main/litgpt/finetune/lora.py#L260) (final), and analogously in `finetune/full.py`, `finetune/adapter.py`, `finetune/adapter_v2.py`, and `pretrain.py` (all 5 scripts).
- These were added in [PR #1228 "Print initial validation loss + final validation loss"](https://github.com/Lightning-AI/litgpt/pull/1228) (commit [fe1374c](https://github.com/Lightning-AI/litgpt/commit/fe1374c)), which is the literal "should be easy for me to add" follow-up referenced in this issue.

`final_validation` defaults to `True` so users get the end-of-training loss out of the box; `initial_validation` defaults to `False` (the sanity-check path still runs 2 iters via the `else` branch at e.g. `lora.py:315`), which feels like a sensible balance for the original "small-dataset confusion" symptom — bumping it to `True` is a one-flag opt-in.

If I'm missing something — e.g. a script where this got missed, or a follow-up doc that should be added — point me at it and I'll dig further. Otherwise would you mind closing this out?

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against current `main`.*
